lib/AST/ASTContext.cpp

```diff
@@ -2234,12 +2243,17 @@ ModuleDecl *ASTContext::getOverlayModule(const FileUnit *FU) {
       return Existing;
   }
 
+  if (PreModuleImportCallback) {
+    SmallString<256> path;
```

Comment:
This function will likely be relatively high up in the call stack, so I wouldn't allocate such a large local variable here. Maybe reduce it to something like 16 or factor this into a helper function?